internal/provider/resources/deployment_test.go

```diff
@@ -363,6 +372,187 @@ func TestAccResource_deployment(t *testing.T) {
 	})
 }
 
+//nolint:paralleltest // we use the resource.ParallelTest helper instead
+func TestAccResource_deployment_global_concurrency_limit(t *testing.T) {
+	workspace := testutils.NewEphemeralWorkspace()
+	deploymentName := testutils.NewRandomPrefixedString()
+	flowName := testutils.NewRandomPrefixedString()
+
+	parameterOpenAPISchema := `{"type": "object", "properties": {"some-parameter": {"type": "string"}}}`
+	expectedParameterOpenAPISchema := testutils.NormalizedValueForJSON(t, parameterOpenAPISchema)
+
+	parameterOpenAPISchemaUpdate := `{"type": "object", "properties": {"some-parameter": {"type": "string"}, "some-other-parameter": {"type": "string"}}}`
+	expectedParameterOpenAPISchemaUpdate := testutils.NormalizedValueForJSON(t, parameterOpenAPISchemaUpdate)
+
+	cfgCreate := deploymentConfig{
+		DeploymentName:         deploymentName,
+		FlowName:               flowName,
+		DeploymentResourceName: fmt.Sprintf("prefect_deployment.%s", deploymentName),
+		Workspace:              workspace.Resource,
+		WorkspaceIDArg:         workspace.IDArg,
+
+		CollisionStrategy:        "ENQUEUE",
+		Description:              "My deployment description",
+		EnforceParameterSchema:   true,
+		Entrypoint:               "hello_world.py:hello_world",
+		GlobalConcurrencyLimitID: true,
```

Comment:
Could we move this into the existing tests above? I don't think we need to duplicate most of the other testing setup here just for concurrency limit ID, right?